From c1ee013288501b3d366833062aea71311d6cff8f Mon Sep 17 00:00:00 2001
From: Gerrit Pape <pape@smarden.org>
Date: Tue, 20 Apr 2010 19:31:23 +0000
Subject: [PATCH 13/13] sparc: use -mcpu=v9 instead of -mcpu=supersparc

This fixes a build failure of libowfat on sparc:

diet -v -Os gcc -g -o t t.o libowfat.a `cat libsocket` -lpthread
gcc -nostdlib -static -L/usr/lib/diet/lib-sparc /usr/lib/diet/lib-sparc/start.o -g -o t t.o libowfat.a -lpthread -isystem /usr/lib/diet/include -D__dietlibc__ -Os -mcpu=supersparc /usr/lib/diet/lib-sparc/libc.a -lgcc /usr/lib/diet/lib-sparc/libc.a
/usr/lib/diet/lib-sparc/libc.a(vprintf.o): In function `vprintf':
vprintf.c:(.text+0x1c): warning: warning: the printf functions add several kilobytes of bloat.
t.o: In function `main':
t.c:(.text+0x30): undefined reference to `__sync_add_and_fetch_4'
t.c:(.text+0x48): undefined reference to `__sync_add_and_fetch_4'
t.c:(.text+0x64): undefined reference to `__sync_bool_compare_and_swap_4'
t.c:(.text+0x80): undefined reference to `__sync_bool_compare_and_swap_4'
collect2: ld returned 1 exit status

According to http://www.debian.org/releases/stable/sparc/ch02s01.html.en
supersparc isn't supported anymore anyway, and multisparc and niagara
are v9.
---
 diet.c             |    2 +-
 sparc/Makefile.add |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/diet.c b/diet.c
index 49bc928..77f7487 100644
--- a/diet.c
+++ b/diet.c
@@ -29,7 +29,7 @@ static const char* Os[] = {
 	 "-falign-functions=1","-falign-jumps=1",
 	 "-falign-loops=1","-fomit-frame-pointer",0,
   "x86_64","-Os",0,
-  "sparc","-Os","-mcpu=supersparc",0,
+  "sparc","-Os","-mcpu=v9",0,
   "sparc64","-Os","-m64","-mhard-quad-float",0,
   "alpha","-Os","-fomit-frame-pointer",0,
 #ifdef __ARM_EABI__
diff --git a/sparc/Makefile.add b/sparc/Makefile.add
index b478fa7..948e348 100644
--- a/sparc/Makefile.add
+++ b/sparc/Makefile.add
@@ -1,5 +1,5 @@
 
-CFLAGS+=-mcpu=supersparc -Os
+CFLAGS+=-mcpu=v9 -Os
 VPATH:=sparc:syscalls.s:$(VPATH)
 LIBOBJ+=$(OBJDIR)/udiv.o $(OBJDIR)/umul.o $(OBJDIR)/urem.o
 LIBOBJ+=$(OBJDIR)/rem.o $(OBJDIR)/sdiv.o
-- 
1.7.0.3

